File: objetos/MaquinaAgricola.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package objetos;

/**
 *
 * @author anacs
 */
public class MaquinaAgricola {
    private String tipo;
    private String modelo;
    private boolean ligada;
    private double combustivel;
    
    public MaquinaAgricola(String tipo, String modelo) {
        this.tipo = tipo;
        this.modelo = modelo;
        this.ligada = false;
        this.combustivel = 100.0;
    }
    
    public void ligar() {
        if (combustivel > 0) {
            ligada = true;
            System.out.println("Máquina " + tipo + " ligada.");
        } else {
            System.out.println("Sem combustível para ligar a máquina.");
        }
    }
    
    public void desligar() {
        ligada = false;
        System.out.println("Máquina " + tipo + " desligada.");
    }
    
    public void operar() {
        if (ligada && combustivel > 0) {
            combustivel -= 10;
            System.out.println("Operando " + tipo + ". Combustível restante: " + combustivel + "%");
        } else {
            System.out.println("Máquina não pode operar.");
        }
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public boolean isLigada() {
        return ligada;
    }

    public double getCombustivel() {
        return combustivel;
    }

    public void setCombustivel(double combustivel) {
        this.combustivel = combustivel;
    }
}

File: python/maquina_agricola.py
class MaquinaAgricola:
    def __init__(self, tipo, modelo):
        self.tipo = tipo
        self.modelo = modelo
        self.ligada = False
        self.combustivel = 100.0

    def ligar(self):
        if self.combustivel > 0:
            self.ligada = True
            print(f"Máquina {self.tipo} ligada.")
        else:
            print("Sem combustível para ligar a máquina.")

    def desligar(self):
        self.ligada = False
        print(f"Máquina {self.tipo} desligada.")

    def operar(self):
        if self.ligada and self.combustivel > 0:
            self.combustivel -= 10
            print(f"Operando {self.tipo}. Combustível restante: {self.combustivel}%")
        else:
            print("Máquina não pode operar.")

    def get_tipo(self):
        return self.tipo

    def set_tipo(self, tipo):
        self.tipo = tipo

    def get_modelo(self):
        return self.modelo

    def set_modelo(self, modelo):
        self.modelo = modelo

    def is_ligada(self):
        return self.ligada

    def get_combustivel(self):
        return self.combustivel

    def set_combustivel(self, combustivel):
        self.combustivel = combustivel